### ISSUE TITLE
Can we include Keycloak as Auth in the Caddy Plugin?

### ISSUE THREAD

**Issue by UNRA1DUser** [opnsense/plugins#4303]
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

- [X] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md
- [X] I have searched the existing issues, open and closed, and I'm convinced that mine is new.
- [X] When the request is meant for an existing plugin, I've added its name to the title.

**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
Can we include Keycloak as Auth in the Caddy Plugin?

**Describe alternatives you've considered**
-

**Additional context**
-

**Comment by Monviech** [comment]
Somebody has to contribute that just like here:

https://github.com/opnsense/plugins/pull/4104

Its probably easier to set up Authelia and use that with caddy.

**Comment by UNRA1DUser** [comment]
Or maybe is it possible to add "oauth2-proxy" to it? So a lot of other Auth. could be configured than.

**Comment by Monviech** [comment]
Do you mean this?

https://github.com/oauth2-proxy/oauth2-proxy/issues/2485#issuecomment-1931230207

**Comment by UNRA1DUser** [comment]
Yes, I meant that one. Sorry for not providing so much information. 



Oh is that also working with authelia? I didn't know that. So I could use authelia or authentik to authenticate with keycloak?

Sounds interesting. I will google it.

**Comment by Monviech** [comment]
You can probably do something like OIDC but its not my expertise.

Both Keycloak and Authelia should support OIDC, so one could be the client of the other.
